### DISCUSSION TITLE
LGT8F328P Software reset

### DISCUSSION THREAD

**Discussion by izzzWERg** [dbuezas/lgt8fx#218] [Q&A] [answered]
Hello!

I am working on migrating my current project from arduino Nano to LogicGreen boards. I have managed almost everything to work properly with LG except one last thing: I need to add an option to programmable reset the board from the sketch. Code that works fine with Nano (and other AVRs)
`void(* resetFunc) (void) = 0; //Programmatically reset the arduino` 
does not work on LGT8Fx. Are there a way that I can use? Thank you for the help.

**Comment by LaZsolt** [comment, marked as answer]
I found some explanations about software reset in the databook: https://raw.githubusercontent.com/dbuezas/lgt8fx/master/docs/LGT8FX8P_databook_v1.0.5-English.pdf

"Software reset: LGT8FX8P has a reset register specially triggered by software. User can reset MCU via this
software."

"Software Reset is triggered via operation at 6th bit of VDTCR register."

![kép](https://user-images.githubusercontent.com/60486174/182852308-d5ecc768-2e8a-45dc-8b62-0976be1ce99f.png)

**Comment by dbuezas** [comment]
Jumping to the zero address should actually work too. I wonder if the bootloader gets confused somehow because some register is in the wrong state
![image](https://user-images.githubusercontent.com/777196/182864215-b351a789-c852-4597-99bf-9e2ac81c0ce4.png)

If the suggestion from @LaZsolt doesn't work either, here are a couple of extra ways:
* Jumping via inline assembly, and
* Forcing the watchdog to trigger

https://github.com/qub1750ul/Arduino_SoftwareReset/blob/master/src/SoftwareReset.hpp

**Comment by izzzWERg** [comment]
Thank you very much for the reply. Operating with VDTCR register works very well. Here is the reset function code if someone needs it:
```
  void resetLG(void){
    VDTCR|=0b10000000;
    VDTCR&=0b10111111;
  }
```
